crates/query-engine/sql/src/sql/convert.rs

```diff
@@ -150,38 +150,53 @@ impl Insert {
         sql.append_identifier(&self.schema.0);
         sql.append_syntax(".");
         sql.append_identifier(&self.table.0);
-        sql.append_syntax("(");
-        for (index, column_name) in self.columns.iter().enumerate() {
-            sql.append_identifier(&column_name.0.to_string());
-            if index < (self.columns.len() - 1) {
-                sql.append_syntax(", ");
-            }
-        }
-        sql.append_syntax(")");
 
-        sql.append_syntax(" VALUES ");
-
-        for (index, object) in self.values.iter().enumerate() {
+        if let Some(columns) = &self.columns {
             sql.append_syntax("(");
-            for (index, value) in object.iter().enumerate() {
-                value.to_sql(sql);
-                if index < (object.len() - 1) {
+            for (index, column_name) in columns.iter().enumerate() {
+                sql.append_identifier(&column_name.0.to_string());
+                if index < (columns.len() - 1) {
                     sql.append_syntax(", ");
                 }
             }
```

Comment:
I think this is clearer if we split it into "first" and "rest". Minor suggestion, feel free to disagree.

```suggestion
            let mut columns_iter = columns.iter();
            if let Some(column_name) = columns_iter.next() {
                sql.append_identifier(&column_name.0.to_string());
            }
            for column_name in columns_iter {
                sql.append_syntax(", ");
                sql.append_identifier(&column_name.0.to_string());
            }
```

Reply:
This pattern is used throughout the module so I'd rather stay consistent. I also find it a bit clearer? Maybe it's because I'm used to it.